monkey/common/event_serializers/pydantic_event_serializer.py

```diff
@@ -0,0 +1,28 @@
+import logging
+from typing import Type
+
+from common.base_models import InfectionMonkeyBaseModel
+from common.events import AbstractAgentEvent
+
+from . import IEventSerializer, JSONSerializable
+
+logger = logging.getLogger(__name__)
+
+
+class PydanticEventSerializer(IEventSerializer):
```

Comment:
You can make `PydanticEventSerializer` a generic, and then use the generic type for all of the type hints instead of `AbstractAgentEvent`.

You'll need `T = TypeVar('T', bound=AbstractAgentEvent)`

See https://docs.python.org/3/library/typing.html#generics
and https://docs.python.org/3/library/typing.html#typing.TypeVar